goconvey.go

```diff
@@ -45,15 +47,66 @@ func flags() {
 	flag.StringVar(&excludedDirs, "excludedDirs", "vendor,node_modules", "A comma separated list of directories that will be excluded from being watched.")
 	flag.StringVar(&workDir, "workDir", "", "set goconvey working directory (default current directory).")
 	flag.BoolVar(&autoLaunchBrowser, "launchBrowser", true, "toggle auto launching of browser.")
+	flag.BoolVar(&cleanupTmpFolder, "cleanupTemp", false, "cleanup tmp folder with reports and static web files")
 
 	log.SetOutput(os.Stdout)
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 }
 func folders() {
 	_, file, _, _ := runtime.Caller(0)
 	here := filepath.Dir(file)
-	static = filepath.Join(here, "/web/client")
-	reports = filepath.Join(static, "reports")
+	localStatic := filepath.Join(here, "web", "client")
+
+	currentUser, err := user.Current()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	tmpDest := filepath.Join(os.TempDir(), fmt.Sprintf("/goconvey-%v", currentUser.Uid))
+
+	mkdir := func(folder string) {
+		if err := os.MkdirAll(folder, 0700); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	cleanupAndMkdir := func(folder string) {
+		if stat, err := os.Stat(folder); err == nil && stat != nil {
+			if !stat.IsDir() {
+				if err = os.Remove(folder); err != nil {
+					log.Fatal(err)
+				}
+
+				mkdir(folder)
+			} else {
+				if cleanupTmpFolder {
+					if err = os.RemoveAll(folder); err != nil {
+						log.Fatal(err)
+					}
+
+					mkdir(folder)
+				}
+			}
+		} else {
+			if err != nil && stat == nil {
+				mkdir(folder)
+			}
+		}
+	}
+
+	tmpStaticPath := filepath.Join(tmpDest, "web", "client")
+	cleanupAndMkdir(tmpStaticPath)
+	tmpReportsPath := filepath.Join(tmpDest, "reports")
+	cleanupAndMkdir(tmpReportsPath)
+
+	if err := copy.Copy(localStatic, tmpStaticPath); err != nil {
+		log.Fatal(err)
+	}
+
+	static = tmpStaticPath
```

Comment:
i.e. why can't 'static' continue to be localStatic?

Reply:
Ah, is this because this static route path would normally serve the files, too? If that's the case, could we just add another route to the http router for the "reports" subdirectory of the tempdir?

Reply:
Er, nevermind... I see that you add another route for reports already...

Reply:
> could we just add another route to the http router for the "reports" subdirectory of the tempdir?
File handler will get /reports/*.html path from http requests only if we'll use `tmpDir` root instead of `reports` dir  :\
i.e. If I'll use `reports` var it'll get `/reports/reports/*` path and lead to 404, tldr; I didn't want to play with url rewritting.

> I see that you add another route for reports already...
Yeah.

> why can't 'static' continue to be localStatic?
Well, I think It actually can, let me double check.
As far as I remember there was some issue with coverage progress bar, might be mixing up things...